libraries/chain/protocol/transaction.cpp

```diff
@@ -262,21 +262,26 @@ struct sign_state
 };
 
 
-void verify_authority( const vector<operation>& ops, const flat_set<public_key_type>& sigs,
-                       const std::function<const authority*(account_id_type)>& get_active,
-                       const std::function<const authority*(account_id_type)>& get_owner,
-                       bool allow_non_immediate_owner,
-                       uint32_t max_recursion_depth,
-                       bool  allow_committe,
-                       const flat_set<account_id_type>& active_aprovals,
-                       const flat_set<account_id_type>& owner_approvals )
+void verify_authority(const vector<operation>& ops, const flat_set<public_key_type>& sigs,
+                      const std::function<const authority*(account_id_type)>& get_active,
+                      const std::function<const authority*(account_id_type)>& get_owner,
+                      bool allow_non_immediate_owner,
+                      bool ignore_custom_operation_required_auths,
+                      uint32_t max_recursion_depth,
+                      bool  allow_committe,
+                      const flat_set<account_id_type>& active_aprovals,
+                      const flat_set<account_id_type>& owner_approvals)
 { try {
    flat_set<account_id_type> required_active;
    flat_set<account_id_type> required_owner;
    vector<authority> other;
 
-   for( const auto& op : ops )
-      operation_get_required_authorities( op, required_active, required_owner, other );
+   for (const auto& op : ops) {
+      if (op.which() == operation::tag<custom_operation>::value && ignore_custom_operation_required_auths)
```

Comment:
This distinction should be moved into `operation_get_required_authorities`. Affects `proposal.required_active_approvals`, account history and impacted_accounts (db_notify.cpp).

Reply:
Speaking of db_notify, `get_impacted_account_visitor::void operator()( const custom_operation& op )` should be modified as well